src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -76,26 +75,56 @@ public static IBrighterBuilder UseExternalOutbox(this IBrighterBuilder brighterB
         {
             if (outbox is IAmAnOutboxSync<Message>)
             {
-                brighterBuilder.Services.Add(new ServiceDescriptor(typeof(IAmAnOutboxSync<Message>), _ => outbox,
-                    ServiceLifetime.Singleton));
+                brighterBuilder.Services.Add(new ServiceDescriptor(typeof(IAmAnOutboxSync<Message>), _ => outbox, ServiceLifetime.Singleton));
             }
 
             if (outbox is IAmAnOutboxAsync<Message>)
             {
-                brighterBuilder.Services.Add(new ServiceDescriptor(typeof(IAmAnOutboxAsync<Message>), _ => outbox,
-                    ServiceLifetime.Singleton));
+                brighterBuilder.Services.Add(new ServiceDescriptor(typeof(IAmAnOutboxAsync<Message>), _ => outbox, ServiceLifetime.Singleton));
             }
 
             return brighterBuilder;
              
         }
 
+         /// <summary>
+         /// Uses an external Brighter Inbox to record messages received to allow "once only" or diagnostics (how did we get here?)
+         /// Advantages: by using an external inbox then you can share "once only" across multiple threads/processes and support a competing consumer
+         /// model; an internal inbox is useful for testing but outside of single consumer scenarios won't work as intended
+         /// If not null, registers singletons with the service collecion :-
+         ///  - IAmAnInboxSync - what messages have we received
+         ///  - IAmAnInboxAsync - what messages have we received (async pipeline compatible)
+         /// </summary>
+         /// <param name="brighterBuilder">Extension method to support a fluent interface</param>
+         /// <param name="inbox">The external inbox to use</param>
+         /// <param name="inboxConfiguration">If this is null, configure by hand, if not, will auto-add inbox to handlers</param>
+         /// <returns></returns>
+         public static IBrighterBuilder UseExternalInbox(this IBrighterBuilder brighterBuilder, IAmAnInbox inbox, InboxConfiguration inboxConfiguration = null)
```

Comment:
While I see we have UseExternalOutbox form what I can see in the code its not used anyway and instead we appear to be using the Specific ones (UseInMemoryOutbox / UseMsSqlOutbox / UsePostgreSqlOutbox / ect.) 

I am a fan of having the specific ones as its a little easier to read, however I see That when I put in UseMsSqlOutbox I added a reference from Paramore.Brighter.Outbox.MsSql to Paramore.Brighter.DependencyInjection meaning that if you want to use MsSql outbox and an Microsoft DI library you are going to have to pull in the Microsoft DI DLL.

Reply:
In this case in particular it reflects that in sample code I use Sqlite for development and MySQL for Production. Now, I know some folks don't like that, but... that kind of scenario is much easier if you don't have a specific override